.github/workflows/checks.yaml

```diff
@@ -29,6 +29,18 @@ jobs:
         version: v2.0.2
         args: --timeout 3m --verbose cmd/... pkg/...
 
+    - name: Extract branch name
+      shell: bash
+      run: |
+        # Extract branch name and sanitize it for Docker tag compatibility
+        if [[ "${{ github.ref_type }}" == "branch" ]]; then
+          BRANCH_NAME="${{ github.ref_name }}"
+          # Replace invalid characters with dashes and convert to lowercase
+          SANITIZED_BRANCH=$(echo "$BRANCH_NAME" | sed 's/[^a-zA-Z0-9._-]/-/g' | tr '[:upper:]' '[:lower:]')
+          echo "branch_tag=${SANITIZED_BRANCH}" >> $GITHUB_ENV
+        fi
```

Comment:
why do we need this here ? this would only be required on img push workflows right ?